### DISCUSSION TITLE
multiple documentation pages isn't obvious from UI

### DISCUSSION THREAD

**Discussion by heckj** [SwiftPackageIndex/SwiftPackageIndex-Server#2022] [Bug]
**Please describe the bug**

Related to https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/issues/1974, it's not obvious that the documentation headers are pull down menus that can lead to multiple modules within a package.

Please give a clear and concise description of what went wrong.

For example, @MaxDesiatov was looking at the JavascriptKit project and didn't realize that the final element was a pull down:
I'm wondering if a disclosure triangle (or equivalent kind of UX affordance) wouldn't make sense there to provide a hint

**Explain the steps needed to reproduce the bug**

1. Navigate to [to 'JavaScriptKit docs'](https://swiftpackageindex.com/swiftwasm/JavaScriptKit/main/documentation/javascriptbigintsupport)

**What was the expected behaviour?**

Max didn't realize that final element was a pull-down menu, and thought documentation was missing.

**Screenshots**

<img width="882" alt="Screen Shot 2022-08-17 at 10 32 57 AM" src="https://user-images.githubusercontent.com/43388/185205620-da085af8-6ab5-45da-be77-9e20bfc9285a.png">

If applicable, please add screenshots to help explain your problem.

**Comment by daveverwer** [comment]
Apologies for the delay in getting to this!

I suspected as soon as you suggested it that a disclosure triangle would be a bit too much, given that we need to disclose *two* dropdowns, not just one. Putting another triangle next to the breadcrumb triangles, and it is:

![Web 1920 – 1](https://user-images.githubusercontent.com/5180/189680872-d29ad500-3c3e-458a-88f1-80f13bd2b7eb.png)

Even if we played with colour/transparency, I think this many triangles is always going to be too much.

Next, I played with ellipses instead of triangles, but it’s a bit weird:

![Web 1920 – 3](https://user-images.githubusercontent.com/5180/189680991-3e1ed73f-25ba-477e-bf3a-69cacc3f7e29.png)

Next, the drop-down on the archive is *much* more important than the one for the branch, which doesn’t hurt anyone if it’s not discovered. I don’t like making it inconsistent, but we could potentially do something only for the archive dropdown:

![Web 1920 – 2](https://user-images.githubusercontent.com/5180/189680924-206a7f0b-4334-45c0-9fad-e5ea52d331ed.png)

Then, I came up with something that was a little more promising:

![Web 1920 – 4](https://user-images.githubusercontent.com/5180/189681037-f280377a-dd55-4215-9011-350fb87355a7.png)

The biggest issue here is mobile. It’s possible we could leave things as they are on mobile, but that’s then two different breadcrumb bars depending on the device, which is not ideal at all.

All discussion on these potential fixes to this is welcome, as well as new ideas! However, until we have a path forward, I’m going to move this to be a discussion rather than an issue.

**Reply by finestructure** [reply]
I like the double disclosure triangle the best. I feel it clearly communicates what's going to happen, what the hierarchy is.
